### ISSUE TITLE
Literal strings and arrays do not allow methods

### ISSUE THREAD

**Issue by bunglegrind** [douglascrockford/JSLint#489]
**Describe the bug**
1. A clear and concise description of what the bug is.

As the title stated, jslint raises an error when methods are invoked from a literal entity (I haven't checked on objects, though)

2. Jslint edition:

Latest

3. Code snippet (if applicable):

```javascript
/*jslint devel*/
"Hi".includes("a");
[0, 1, 2].includes(2);
```

**Expected behavior**

No errors

**Comment by rob-legrand** [comment]
You can see which methods are allowed to be called on string and array literals by looking at the `infix_dot` and `infix_option_chain` functions in `jslint.mjs`.  It seems that `indexOf` and `repeat` can be called on strings and that `concat`, `flat`, `flatMap`, `forEach`, `join` and `map` can be called on arrays (though `flat` and `flatMap` don't seem to be allowed after `?.`; shouldn't `.` and `?.` be consistent with each other?).  I like the idea of also allowing `includes` for both strings and arrays.  I'd also choose to allow `match` for strings and `some`, `every`, `filter`, `reduce`, `toSorted`, `toReversed`, `find` and `findIndex` for arrays (I often build new arrays using spread syntax, so I end up calling methods on literal arrays often).

**Comment by kaizhu256** [comment]
- yea, i personally wrap them around identity-function, as hacky workaround ^^;;;
```js
function noop(arg0) {
// This function does nothing, and returns first argument.
    return arg0;
}
noop("Hi").includes("a");
noop([0, 1, 2]).includes(2);
```

- happy to accept pull-request with latest string / array methods.

**Comment by rob-legrand** [comment]
[bunglegrind](https://github.com/bunglegrind), will you take this one?  If you'd rather not, I'll be glad to do it.

**Comment by bunglegrind** [comment]
> [bunglegrind](https://github.com/bunglegrind), will you take this one? If you'd rather not, I'll be glad to do it.

do as you please!

BTW I agree with you: `.` and `?.` whitelists must be consistent with each other.

**Comment by rob-legrand** [comment]
I propose allowing the following methods to be called on string literals:
- `charCodeAt`
- `includes`
- `indexOf`
- `match`
- `repeat`
- `replace`
- `toLowerCase`
- `toUpperCase`

I propose allowing the following methods to be called on array literals:
- `every`
- `filter`
- `find`
- `findIndex`
- `flat`
- `flatMap`
- `forEach`
- `includes`
- `join`
- `map`
- `reduce`
- `some`
- `toReversed`
- `toSorted`

Thoughts?  Suggestions?  Objections?

**Comment by kaizhu256** [comment]
- personally, I'm ok allowing most non-deprecated legal array/string methods added since es6
- mainly b/c i'm not knowledgeable about footguns in this area, so can't argue against them ^^;;;

**Comment by kaizhu256** [comment]
fixed in #494
